libs/community/langchain_community/vectorstores/lancedb.py

```diff
@@ -52,25 +57,41 @@ def __init__(
                 "Could not import lancedb python package. "
                 "Please install it with `pip install lancedb`."
             )
-        self.lancedb = lancedb
         self._embedding = embedding
         self._vector_key = vector_key
         self._id_key = id_key
         self._text_key = text_key
         self._table_name = table_name
+        self.api_key = api_key
+        self.region = region
+        self.mode = mode
+
+        if os.getenv("LANCE_API_KEY") is not None:
+            api_key = os.getenv("LANCE_API_KEY")
+
+        if "db://" in uri and api_key is None:
```

Comment:
self.api_key?

Reply:
Its not needed because I am not passing the api key in the class object the value is being reused in the function only not anywhere else, its abstracted only connection, uri and table name can be
Accessed by user

Reply:
Will change if you want a specific design